File: APIExamples/CEP_API/dotnet/SubscriptionOperations.cs
using ASUA.CEP.SDK.Api;
using ASUA.CEP.SDK.Client;
using ASUA.CEP.SDK.Model;
using System;
using System.Collections.Generic;

namespace ASUAWorkShop_APIExamples.CEP_API
{
    class SubscriptionExamples
    {
        ApiClient apiClient = new ApiClient("http://netigma.netcad.com.tr/cep");
        SubscriptionApi subscriptionApi = null;

        public SubscriptionExamples()
        {
            Configuration conf = new Configuration(apiClient);
            conf.AddApiKey("Authorization", "YOUR_API_KEY");

            subscriptionApi = new SubscriptionApi(conf);
        }


        public void RetrieveSubscriptions()
        {
            //Get single subscription by it's key
            int subscriptionKey = 123;
            SubscriptionModel subscriptionModel = subscriptionApi.GetSubscription(subscriptionKey);

            Console.WriteLine("Subscription Key : " + subscriptionModel.Id);
            Console.WriteLine(subscriptionModel.ToString());


            //Get all subscriptions
            List<SubscriptionModel> subscriptionList = subscriptionApi.GetAllSubscriptions();

            Console.WriteLine(subscriptionList.Count);
        }

        public void StartSubscriptions()
        {
            //Start single subscription by it's key
            int subscriptionKey = 123;
            Message message1 = subscriptionApi.Start(subscriptionKey);
        }

        public void StopSubscriptions()
        {
            //Stop single subscription by it's key
            int subscriptionKey = 123;
            Message message1 = subscriptionApi.Stop(subscriptionKey);
        }

        public void DeleteSubscriptions()
        {
            //Delete single subscription by it's key
            int subscriptionKey = 123;
            Message message1 = subscriptionApi.Unsubscribe(subscriptionKey);
        }

    }
}

File: APIExamples/CEP_API/javascript/subscriptionOperations.js
var ASUACEPSDK = require('asuacepsdk');

var apiClient = new ASUACEPSDK.ApiClient();
apiClient.basePath = "http://netigma.netcad.com.tr/cep";
apiClient.authentications.Authorization.apiKey = "YOUR_API_KEY";

var subscriptionApi = new ASUACEPSDK.SubscriptionApi(apiClient);

//Name of the rule that will be subscribed
var statementName = "rule_name_123";

var mailSubscription = new ASUACEPSDK.MailSubscription();
mailSubscription.mailAddress = "mail@address.com";
mailSubscription.message = "mail content";
mailSubscription.started = true;

//create mail subscription
subscriptionApi.createMailSubscription(statementName, mailSubscription, function (error, data, response) {
    if (!error) {
        console.log("Mail Subscription created, subscription key : " + data.key)
    }
})


//get all subscriptions
subscriptionApi.getAllSubscriptions(function (error, subscriptionList, response) {
    if (!error) {
        console.log("Total subscription : " + subscriptionDetail.length);
        console.log(JSON.stringify(subscriptionDetail));
    }
})


//get a subscription detail by it's id
var subscriptionKey = 123;
subscriptionApi.getSubscription(subscriptionKey, function (error, subscriptionDetail, response) {
    if (!error) {
        console.log("Subscription : " + JSON.stringify(subscriptionDetail));
    }
})

//start subscription it's id
var subscriptionKey = 123;
subscriptionApi.getSubscription(subscriptionKey, function (error, subscriptionDetail, response) {
    if (!error) {
        console.log("Subscription : " + JSON.stringify(subscriptionDetail));
    }
})







